app/preprints/new-version/route.ts

```diff
@@ -0,0 +1,78 @@
+import Store from '@ember-data/store';
+import Route from '@ember/routing/route';
+import RouterService from '@ember/routing/router-service';
+import { inject as service } from '@ember/service';
+import Toast from 'ember-toastr/services/toast';
+import Intl from 'ember-intl/services/intl';
+
+import Theme from 'ember-osf-web/services/theme';
+import MetaTags, { HeadTagDef } from 'ember-osf-web/services/meta-tags';
+import PreprintModel from 'ember-osf-web/models/preprint';
+
+export default class PreprintNewVersionRoute extends Route {
+    @service store!: Store;
+    @service theme!: Theme;
+    @service router!: RouterService;
+    @service metaTags!: MetaTags;
+    @service toast!: Toast;
+    @service intl!: Intl;
+
+
+    headTags?: HeadTagDef[];
+
+    buildRouteInfoMetadata() {
+        return {
+            osfMetrics: {
+                providerId: this.theme.id,
+            },
+        };
+    }
+
+    async model(args: any) {
+        try {
+            const provider = await this.store.findRecord('preprint-provider', args.provider_id);
+            this.theme.providerType = 'preprint';
+            this.theme.id = args.provider_id;
+
+            const preprint: PreprintModel = await this.store.findRecord('preprint', args.guid);
+
+            // TODO: Re-evaluate if re-route logic is necessary
+            // if (!preprint.canCreateNewVersion) {
+            //     let message = this.intl.t('preprints.submit.new-version.redirect.latest-published',
+            //         { preprintWord: provider.documentType.singular });
+            //     if (!preprint.currentUserIsAdmin) {
+            //         message = this.intl.t('preprints.submit.new-version.redirect.permission',
+            //             { preprintWord: provider.documentType.singular });
+            //     }
+
+            //     const title = this.intl.t('preprints.submit.new-version.redirect.title');
+            //     this.toast.info(message, title);
+            //     this.router.transitionTo('preprints.detail', args.provider_id, args.guid);
+            //     return null;
+            // }
```

Comment:
What do we want to do with this?

Reply:
I think I'll go ahead and remove this bit for now. I think it's best to wait and see if this particular case (users trying to go to the create-version page directly) is going to be an issue with users, and we can discuss with product folks then on how we want to handle it